### PR TITLE
Use ref

### LINKED ISSUE #86: Use Refs for Form Focus & Select
#### Describe the desired behavior
In the Form Validation examples, use React Refs instead of document & DOM to focus input fields.

#### Describe the current behavior
It's currently using javascript.

#### Is this request related to a current issue?
No

#### Additional Context
React Design Pattern Form Validation Examples

### PR DESCRIPTION
**Pull request by huayunh**
Fixes #86

#### Changes proposed in this Pull Request:
- Use `useRef()` where a stage change is involved (`.focus()`, `.select()`), and thus removed all `getElementById` functions
- Changed Cypress tests. Now, `yarn cytest` should yield an all-pass result.
  - `id` seems to make Cypress less likely to time out than `[data-cy]`.
- Cleaned up the code

### CODE REVIEW

**Comment by huayunh** [comment]
(unit tests run fine locally)

File: cypress/integration/form-validation/fix-length-passcode.spec.js [hunk 1]
```diff
@@ -11,40 +10,40 @@ describe('Form validation fix length passcode', () => {
     });
 
     it('should display incorrect passcode error', () => {
-        cy.get('#passcode-input').type('789789')
-        cy.get('#passcode-input-helper-text').should('contain', 'Incorrect Passcode')
+        cy.get('#passcode-input').type('789789');
+        cy.get('.MuiFormHelperText-root.Mui-error').should('contain', 'Incorrect Passcode');
```
Comment (jeffvg) [inline cypress/integration/form-validation/fix-length-passcode.spec.js:14]: #passcode-input-helper-text can be used here since it's exposed.

**Review by jeffvg — APPROVED** [review]
looks good

**Review by joebochill — APPROVED** [review]
Code looks good

**Pull request merged**